### PR TITLE
feat(swagger3): export @Parameter#example as parameter example

### PR DESCRIPTION
**Pull request by jayliaoo**
## Description

Read the example attribute from @Parameter via a new param.demo rule and pass it through the setParam/setFormParam/setPathParam/setHeader/setResponseHeader script APIs (now accepting an optional example argument) so query/form/path/header parameters carry an example value.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Test coverage improvement

## Related Issues

Impletments #1421

## Changes Made

- Read the example attribute from @Parameter via a new param.demo rule and pass it through the setParam/setFormParam/setPathParam/setHeader/setResponseHeader script APIs (now accepting an optional example argument) so query/form/path/header parameters carry an example value.

## Testing

- [x] Unit tests pass
- [x] Manual testing completed
- [x] New tests added for new functionality

## Checklist

- [x] My code follows the project's architecture principles
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: src/main/resources/extensions/swagger3.config [hunk 1]
```diff
@@ -31,6 +31,7 @@ param.ignore=@io.swagger.v3.oas.annotations.Parameter#hidden
 param.required=@io.swagger.v3.oas.annotations.Parameter#required
 param.doc[@io.swagger.v3.oas.annotations.Parameter#deprecated]=「已废弃」
 param.doc=@io.swagger.v3.oas.annotations.Parameter#description
+param.demo=groovy:it.ann("io.swagger.v3.oas.annotations.Parameter","example")?:null
```
Comment (tangcent) [inline src/main/resources/extensions/swagger3.config]: Two concerns with this rule:

1. **Empty-string vs null.** The stub `Parameter.java` declares `String example() default ""`, so for any `@Parameter` without an explicit `example`, `it.ann(...)` returns `""`. Groovy `?:` treats `""` as falsy so `"" ?: null` → `null` — that's the intended behavior, but it's subtle and a future reader (or a change to a non-empty default) will trip on it. Make it explicit, e.g. `def ex = it.ann("io.swagger.v3.oas.annotations.Parameter","example"); return (ex == null || ex.isEmpty()) ? null : ex`.

2. **Do we need the script-side plumbing at all?** `param.demo` is already consumed by `DocMetadataResolver.resolveParamDemo` → `SpringMvcClassExporter.buildSingleParameter` (`SpringMvcClassExporter.kt:329` → `example = demo ?: mock`). For the Spring MVC case this PR's integration test exercises (`OrderController.listOrders`), the example is already populated via `buildSingleParameter` *without* touching `setParam`. The widened `setParam(...,example)` only matters for the `export.after[@io.swagger.v3.oas.annotations.Parameter]` path, which would *add a second param* — which is exactly what the new `testParameterExtractsExample` defensively asserts does *not* happen. That suggests the script path is dead code for this scenario. If so, please drop the `ApiModels.kt` / `ScriptApiEndpoint.kt` / `resolve_parameter` changes and keep just this one-line rule addition; the integration test will still pass via `buildSingleParameter`. If the script path is meant to be live (e.g. for non-Spring frameworks), please cover that case with a separate test and remove the misleading "must not duplicate" assertion here.

File: src/main/resources/extensions/swagger3.config [hunk 2]
```diff
@@ -39,13 +40,13 @@ resolve_parameter=```
...
+        api.setParam(map.name,null,required,desc,map.example)
     }else if(paramType=="form"){
-        api.setFormParam(map.name,"",required,desc)
+        api.setFormParam(map.name,"",required,desc,map.example)
     }else if(paramType=="path"){
-        api.setPathParam(map.name,null,desc)
+        api.setPathParam(map.name,null,desc,map.example)
     }else if(paramType=="header"){
-        api.setHeader(map.name,null,required,desc)
+        api.setHeader(map.name,null,required,desc,map.example)
```
Comment (tangcent) [inline src/main/resources/extensions/swagger3.config:49]: If we keep the setter widening, `map.example` should also be threaded through `resolve_swagger3_api_response` for `setResponseHeader` (currently ~line 83 calls `api.setResponseHeader(name,"",required,desc)` with no example). Either widen consistently or keep the response-header API unchanged in this PR.

File: src/main/kotlin/com/itangcent/easyapi/core/export/ApiModels.kt [hunk 3]
```diff
@@ -42,52 +42,56 @@ data class ApiEndpoint(
     val metadata: ApiMetadata,
     val extensions: Extension = Extension.EMPTY
 ) {
-    fun setParam(name: String?, defaultValue: String?, required: Boolean, desc: String?) {
+    fun setParam(name: String?, defaultValue: String?, required: Boolean, desc: String?, example: String? = null) {
```
Comment (tangcent) [inline src/main/kotlin/com/itangcent/easyapi/core/export/ApiModels.kt:45]: Per the swagger3.config comment — for Spring MVC params (the only case the integration test exercises) this `example` argument has no consumer, because the example is populated earlier by `SpringMvcClassExporter.buildSingleParameter` via `param.demo`. `ApiParameter.example` already exists on master; this PR is only widening the script setter.

If we do keep the script-API widening, a few notes on this hunk:
- Please add a one-line KDoc on each setter explaining the new `example` arg and that it's optional for backward script compatibility. (`@JvmOverloads` is on the `ScriptApiEndpoint` wrapper, not here.)
- `setPathParam` hard-codes `required = true` and now also takes `example` — fine, but worth a KDoc note since the signature shape diverges from the other setters.
- Name consistency: the underlying field is `example`, but the rule key feeding it is `param.demo`. A one-line comment would help future readers connect the dots.

File: src/main/kotlin/com/itangcent/easyapi/core/rule/context/ScriptApiEndpoint.kt [hunk 4]
```diff
@@ -50,20 +50,24 @@ class ScriptApiEndpoint(
         endpoint.description = desc
     }
 
-    fun setParam(name: String?, defaultValue: String?, required: Boolean, desc: String?) {
-        endpoint.setParam(name, defaultValue, required, desc)
+    @JvmOverloads
+    fun setParam(name: String?, defaultValue: String?, required: Boolean, desc: String?, example: String? = null) {
```
Comment (tangcent) [inline src/main/kotlin/com/itangcent/easyapi/core/rule/context/ScriptApiEndpoint.kt:68]: `@JvmOverloads` will generate two overloads per setter (4-arg + 5-arg). Since these are invoked from Groovy scripts, the 4-arg overload is the backward-compat path — good. But please confirm: are these setters ever called from Java? If not, `@JvmOverloads` is unnecessary and just adds bytecode; Groovy picks the loosest matching signature on its own. If yes, the generated overload names need to be considered a public API now.

File: src/test/kotlin/com/itangcent/easyapi/framework/springmvc/config/Swagger3ConfigIntegrationTest.kt [hunk 5]
```diff
@@ -138,6 +138,39 @@ class Swagger3ConfigIntegrationTest : EasyApiLightCodeInsightFixtureTestCase() {
...
+            it.httpMetadata?.method == HttpMethod.GET &&
+            it.httpMetadata?.path?.contains("list") == true
+        }
+        assertNotNull("Should find GET /order/list endpoint", listEndpoint)
+
+        val params = listEndpoint?.httpMetadata?.parameters ?: emptyList()
+        // @Parameter on a method argument is resolved via param.demo (buildSingleParameter),
+        // not via the export.after script — so it must not duplicate the Spring-resolved param.
+        val pageParams = params.filter { it.name == "page" }
+        assertEquals("page parameter should not be duplicated", 1, pageParams.size)
```
Comment (tangcent) [inline src/test/kotlin/com/itangcent/easyapi/framework/springmvc/config/Swagger3ConfigIntegrationTest.kt:158]: This `assertEquals("page parameter should not be duplicated", 1, pageParams.size)` is a smell, not a feature. It's defending against the swagger3 `export.after[@io.swagger.v3.oas.annotations.Parameter]` rule producing a *second* `page` param via the newly-widened `setParam`.

If the script path can't fire here (because Spring MVC resolves the param first and the export.after hook is suppressed/deduped), this assertion is just guarding against a phantom — and the example was already set by `buildSingleParameter`. In that case, the unit under test for `param.demo` is `SpringMvcClassExporter`, not the swagger3 script, and the assertion + comment are misleading.

If the script path *can* fire, this assertion is silently hiding the very duplication bug we should be fixing.

Please either (a) drop the `setParam` example arg + the swagger3 `resolve_parameter` changes and let `buildSingleParameter` carry the example (removing the need for this assertion), or (b) add a positive test that demonstrates *when* the script path fires and how duplication is prevented.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,13 @@ All notable changes to the EasyAPI plugin will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Added
+-  export swagger3 `@Parameter#example` as the query/form/path/header parameter example
```
Comment (tangcent) [inline CHANGELOG.md]: Please drop this hunk. `CHANGELOG.md` is regenerated by the release scripts on version bump; hand-edits cause merge conflicts on every release and get overwritten. The release notes for this feature will be picked up automatically when it ships.

File: .gitignore [hunk 7]
```diff
@@ -1,6 +1,7 @@
 .gradle/
 build/
 .idea/
+*.iml
```
Comment (tangcent) [inline .gitignore]: Adding `*.iml` is unrelated to the `@Parameter#example` feature. Please split into a separate chore PR (or drop) so this PR stays focused. (Also: `*.iml` is typically covered by IDEA's global ignore; worth checking we actually need it in the repo.)

**Review by tangcent — COMMENTED** [review]
Thanks for the PR — exporting Swagger 3 `@Parameter#example` as a parameter example is a nice addition. Main concerns below.

1. **The `ApiModels.kt` / `ScriptApiEndpoint.kt` API widening is questionable for the case the tests actually cover.** `ApiParameter.example` and `ApiHeader.example` already exist on master, and for Spring MVC params the example is *already* populated by `SpringMvcClassExporter.buildSingleParameter` via `param.demo` (`SpringMvcClassExporter.kt:329` → `example = demo ?: mock`). The widened `setParam/setFormParam/...` setters here are only reachable from the swagger3 `export.after[@io.swagger.v3.oas.annotations.Parameter]` rule, which runs *in addition to* the Spring-resolved param — which is why the new `testParameterExtractsExample` had to defensively assert `pageParams.size == 1` ("should not be duplicated"). If the script path genuinely can't fire for Spring MVC params (which the assertion implies), the widened setters have no real consumer and this whole hunk is dead surface. If it *can* fire, you've built a duplication risk that the test is silently papering over. Either way, please narrow scope: just add `param.demo` to `swagger3.config` and let `buildSingleParameter` pick it up — no script-API change is needed for the Spring MVC case the integration test exercises.

2. **`CHANGELOG.md` should not be hand-edited** — it's regenerated by the release scripts. Please revert that hunk.

3. **`.gitignore` — `*.iml`** is unrelated to this feature; please drop it or split into a separate chore PR.

4. **Empty-string vs null in `param.demo`.** The stub annotation declares `String example() default ""`, so `it.ann(...,"example")` returns `""` for any `@Parameter` that doesn't set `example`. Groovy's `?:` does treat `""` as falsy, so `"" ?: null` → `null` happens to work — but it's subtle. Please make the intent explicit (e.g. guard on `isEmpty()`).

5. **Response-header example is inconsistent.** `setResponseHeader` was widened to accept `example`, but `resolve_swagger3_api_response` (around line 83) does *not* pass one. If the goal is end-to-end examples, thread it through consistently; if not, please don't widen `setResponseHeader` in this PR.

6. The PR was opened against an outdated base (`exporter/model/ApiModels.kt` was relocated to `core/export/ApiModels.kt` by `55e444a9`). The merge-from-upstream resolved the textual conflicts — thanks — but please re-verify the merged file content carefully.

**Comment by tangcent** [comment]
Hi @jayliaoo — thanks again for this feature. I pushed two follow-up commits on top of your branch to resolve the outstanding review comments and make the `@Parameter#example` support complete. Your original two commits are preserved (this was a fast-forward, not a force-push).

## What I changed

**1. Dropped the unrelated hunks** (`814e938`)
- Reverted the `CHANGELOG.md` edit (regenerated by release scripts; causes merge conflicts / gets overwritten).
- Reverted the `*.iml` `.gitignore` line (unrelated to this feature; ship in a separate chore PR).

**2. Fully threaded `example` through the script paths** (`1b160d8`)
- **`swagger3.config`** — simplified `param.demo` to the plain rule (the plugin already normalizes the empty-string default to null, so no extra guard is needed). Threaded `responseHeader.example` into `resolve_swagger3_api_response`'s `setResponseHeader` call, so the response-header setter widening you added is now actually exercised (previously widened in signature but never passed an example).
- **`ScriptApiEndpoint.kt`** — added KDoc on each widened setter explaining the optional `example` arg and the `@JvmOverloads` backward-compat rationale (kept `@JvmOverloads` because Groovy's MOP cannot resolve Kotlin's synthetic default-arg bridge, so the generated Java overloads are needed for any 4-arg script).

**3. Positive test for the script path** — added `testMethodLevelParameterExtractsExampleViaScript`, which covers the `export.after[@Parameter]` → `resolve_parameter` → `api.setParam(...)` path. This is the *only* consumer of the widened setter signature — the argument-level `@Parameter` case goes through `param.demo` → `buildSingleParameter` (already covered by `testParameterExtractsExample`). Both placements are now tested positively.

**4. Test-harness fix in `TestConfigReader`** — `fromConfigText` now resolves `${key}` placeholders, mirroring `LayeredConfigReader`. Without this, templated rules like `${resolve_parameter}` reached the rule engine as a literal Groovy syntax error (silently swallowed under `IGNORE_ERROR` event mode), so `export.after` script rules could never fire in integration tests. This unblocks script-path testing generally.

## Verification
All Swagger3 + ScriptApiEndpoint tests pass, plus a full sweep of the `core.rule.*` and `core.config.*` subsystems (no regressions from the `TestConfigReader` change).

Let me know if you'd like any adjustments.
